src/pal/src/include/pal/thread.hpp

```diff
@@ -314,6 +314,10 @@ namespace CorUnix
         DWORD m_dwLwpId;
         pthread_t m_pthreadSelf;        
 
+#if HAVE_MACH_THREADS
+        mach_port_t m_machPortSelf;
```

Comment:
Are you sure it needs a cleanup? I was under an impression that the pthread_mach_thread_np just exposes mach port that is stored in the pthread_t and is bound to it.